packages/@react-spectrum/calendar/src/CalendarCell.tsx

```diff
@@ -60,6 +61,15 @@ export function CalendarCell({state, currentMonth, ...props}: CalendarCellProps)
 
   let nativeDate = useMemo(() => date.toDate(state.timeZone), [date, state.timeZone]);
   let formatted = useMemo(() => dateFormatter.format(nativeDate), [dateFormatter, nativeDate]);
+  
+  const currentCell = ref.current;
+  const previousCell = currentCell?.parentElement?.previousElementSibling?.firstElementChild;
+  const nextCell = currentCell?.parentElement?.nextElementSibling?.firstElementChild;
+
+  const isCellDisabled = (cell: Element): boolean => cell?.classList.value.includes('is-disabled');
```

Comment:
Not sure if this is the ideal way to verify if the sibling cell is disables, I am open to suggestions

Reply:
better to use state if you can
I see there is already a function on the state for checking if the date is disabled
https://github.com/adobe/react-spectrum/pull/2793/files#diff-fd8b93994dbea995798535e4fb0d1323e12e7290667d0329b10ab4209db5199fR199

Reply:
and there are functions for getting the next day and the previous day I believe

Reply:
What happens for this if we're on the first cell or the last cell of a grid? just undefined?

Reply:
You can use state for the current cell, but from what I've seen, each cell has its own state. You can either control the state for the whole calendar or the state for one cell. I don't think you can use the states of two adjacent cells simultaneously (though I might be wrong) 🤔
There are functions for focusing the next day or calculating the next date. However, what I need is a function that returns the state of the next cell (or the effect of the state, like having a certain class), and I didn't find something like that (again, I might be wrong).
And about the last questions: yes, in that case, it returns undefined. Should I add a fallback to false (`... || false`), so that the API would be more consistent?

Reply:
I mean, can't you check it like this?
```
state.isCellDisabled(props.date.subtract({days: 1}))
state.isCellDisabled(props.date.add({days: 1}))
```

code pulled from just below the link I shared, `isPreviousVisibleRangeInvalid` which I feel like you might just be able to use that function directly?

I'm quite sure we do not want to look in the dom for this state though. My suggestion may not pan out, I'm not super familiar with this code.

Reply:
Oh, I see now what you meant and you're 100% right. Thank you, I've updated this logic and it's way cleaner now 👌 https://github.com/adobe/react-spectrum/pull/2793/commits/7d9032cc899482c340b68b2202b942ee39669470